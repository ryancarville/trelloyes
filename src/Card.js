import React from 'react';
import './Card.css';

function Card(props) {
	return (
		<div className='Card'>
			<h3>{props.title}</h3>
			<p>{props.content}</p>
			<button type='button'> delete </button>
		</div>
	);
}

export default Card;
