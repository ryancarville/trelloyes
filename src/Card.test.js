import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Card Component', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Card />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Card UI renders as expected', () => {
		const props = [{ title: 'First card', content: 'lorem ipsum' }];
		const tree = renderer
			.create(<Card title={props.title} content={props.content} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
