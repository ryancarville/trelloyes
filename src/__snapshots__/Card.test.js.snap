// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Card UI renders as expected 1`] = `
<div
  className="Card"
>
  <h3 />
  <p />
  <button
    type="button"
  >
     delete 
  </button>
</div>
`;
